ibu-imager/ops/ops.go

```diff
@@ -67,3 +79,85 @@ func (o *ops) RunBashInHostNamespace(command string, args ...string) (string, er
 	args = append([]string{command}, args...)
 	return o.RunInHostNamespace("bash", "-c", strings.Join(args, " "))
 }
+
+// WaitForEtcd waits for the availability of an etcd server at the given endpoint.
+// It continuously checks for the existence of a specific key in etcd to determine the server's availability.
+// If the etcd server becomes accessible, it returns nil, indicating success. Otherwise, it returns an error.
+func (o *ops) WaitForEtcd(endpoint string) error {
+	// Create an etcd client configuration with the specified endpoint and timeout
+	etcdConfig := client.Config{
+		Endpoints:   []string{endpoint},
+		DialTimeout: 5 * time.Second,
+	}
+
+	// Initialize the etcd client
+	etcdClient, err := client.New(etcdConfig)
+	if err != nil {
+		return err
+	}
+	defer etcdClient.Close()
+
+	// Create a context with a timeout to limit the waiting duration
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	// Continuously attempt to retrieve a specific key from etcd
+	for {
+		_, err = etcdClient.Get(ctx, "some_key")
+		if err == nil {
+			// Etcd server is up and responding, exit the loop
+			break
+		}
+
+		// Wait for a short interval before retrying
+		time.Sleep(1 * time.Second)
+	}
+
+	// Etcd server is accessible, return nil to indicate success
+	return nil
+}
+
+// GetImageFromPodDefinition reads a YAML file containing pod configuration data
+// and extracts the image name for a given container named.
+// It returns the image name if found, or an error if not found or encountered any issues.
+func (o *ops) GetImageFromPodDefinition(containerImage string) (string, error) {
+
+	// Define a struct to match the structure of the YAML data
+	type PodConfig struct {
```

Comment:
why not corev1.Pod ?

Reply:
huge dependency

Reply:
I agree with @omertuc on this, I'd advocate keeping it lighter for now, if that makes sense